### ISSUE TITLE
New ZIM request: The Cutting Room Floor (TCRF)

### ISSUE THREAD

**Issue by Phantop** [openzim/zim-requests#345]
- Website URL: https://tcrf.net
- License: **CC-BY-3.0**
- Desired ZIM Title: **The Cutting Room Floor**
- Desired ZIM Description: **The Cutting Room Floor is a site dedicated to unearthing and researching unused and cut content from video games. From debug menus, to unused music, graphics, enemies, or levels, many games have content never meant to be seen by anybody but the developers — or even meant for everybody, but cut due to time/budget constraints.**
- Desired ZIM Icon –png (URL or attach one): https://tcrf.net/images/c/c4/SMB2-smiley.png
- Language (ISO 639-3): **eng**
- Desired Main Page (homepage, if different from website URL): **https://tcrf.net/The_Cutting_Room_Floor**
- Is this a MediaWiki?: **yes**

**Comment by RavanJAltaie** [comment]
Requested.
The file will work after the mediawiki mwoffliner be updated.
https://farm.openzim.org/recipes/the_cutting_room_floor

**Comment by benoit74** [comment]
Just request [the task](https://farm.openzim.org/pipeline/4dbfaefd-5e19-415e-a0e3-7d5d1ac81ec0).

**Comment by benoit74** [comment]
Still failing, I will work on it locally first.

**Comment by benoit74** [comment]
@kelson42 : 
I'm not an MW expert at all but looking at https://tcrf.net/api.php, it seems that there is very few modules available. Any chance we might be able to scrape this wiki? 
Is this because the site owner disabled internal modules or is it linked to the fact that this MW is very old?

**Comment by benoit74** [comment]
@VadimKovalenkoSNF maybe you could help us on this one as well. Do you think there is any appropriate settings (with mwoffliner 1.3) which could make the scrapper succeed on this wiki? Would we have more success once mwoffliner 1.4 is out?

**Comment by VadimKovalenkoSNF** [comment]
@benoit74  At the moment there is no renderer that can parse **tcrf** wiki. This wiki doesn't support Mediawiki REST API (`/rest.php`), Wikimedia REST API (`page/html` and `page/mobile-html`) or VisualEditor (`action=visualeditor`), see https://tcrf.net/Special:Version

It can be scraped only by using Action API (`action=parse`) which mwoffliner uses now only to get modules.

Check this example (note, that there is no `w/` or `wiki/` prefix there):
https://tcrf.net/api.php?action=parse&format=json&prop=text%7Cmodules%7Cjsconfigvars%7Cheadhtml&parsoid=1&formatversion=2&page=Final_Fantasy_VII

**Comment by benoit74** [comment]
Thank you for the confirmation !

I will mark this ticket as "upstream" since we will have to wait for `mwoffliner` to support Action API to get articles and/or wiki to be updated.

For my knowledge, could you please correct below statements me if I'm wrong:
- VisualEditor is an extension which comes with Mediawiki but needs to be installed
- Wikimedia REST API is available only on Wikimedia wikis
- Mediawiki REST API is included with MediaWiki 1.35+ 
- Mediawiki Action API is included in all MediaWiki

**Comment by VadimKovalenkoSNF** [comment]
@benoit74 yes, that's right. Here is the Wiki API doc reference just in case: https://www.mediawiki.org/wiki/API:REST_API

**Comment by benoit74** [comment]
@Phantop are you anyhow related to TCRF wiki? do you have any way to ask them if they plan to update MediaWiki to 1.35+ or to install the VisualEditor extension?

**Comment by Phantop** [comment]
> @Phantop are you anyhow related to TCRF wiki? do you have any way to ask them if they plan to update MediaWiki to 1.35+ or to install the VisualEditor extension?

I'm unfortunately not connected with TCRF in any way, I just wanted a local copy of the site when I opened this issue.

**Comment by benoit74** [comment]
No worries

**Comment by benoit74** [comment]
Tried again with new 1.15 and ActionParse renderer, but it fails at fetch website metadata ...

https://farm.openzim.org/pipeline/f8fdc7d0-dd24-4fc2-a6cb-ae2889c932b7

https://tcrf.net/api.php?action=query&meta=siteinfo&format=json&formatversion=2&siprop=general%7Cnamespaces%7Cstatistics%7Cvariables%7Ccategory%7Cwikidesc

**Comment by kelson42** [comment]
Retried again, but the problem is that the MediaWiki is behind a WAF. See https://farm.openzim.org/pipeline/cd7a85f6-9cfc-4ec5-b62f-d97b4a16a464

**Comment by Popolechien** [comment]
Started a new run using 1.17.5 but it failed again. https://farm.openzim.org/pipeline/ae863a49-d0ef-4175-8ab1-bbd9d6029a6f/debug
